src/ServiceBusSQSChannel.php

```diff
@@ -50,21 +59,61 @@ public function send($notifiable, Notification $notification)
             return;
         }
 
-        $message = $notification
-            ->toServiceBus($notifiable)
-            ->getParams();
+        if (!isset($params['from'], $params['events'][0])) {
+            Log::error('Invalid message structure', ['params' => $params]);
+            return;
+        }
+
+        $queueUrl = Arr::get($this->config, 'sqs.queue_url');
+        $isFifoQueue = strpos($queueUrl, '.fifo') !== false;
 
-        $response = $this->sqs->sendMessage([
-            'QueueUrl' => Arr::get($this->config, 'sqs.queue_url'),
-            'MessageBody' => json_encode($message),
-            'MessageGroupId' => $message['from'],
-        ]);
+        $payloadSqs = [
+            'QueueUrl' => $queueUrl,
+            'MessageBody' => json_encode($params),
+        ];
 
-        $event = $message['events'][0];
+        if ($isFifoQueue) {
+            $payloadSqs['MessageGroupId'] = $params['from'];
+            $payloadSqs['MessageDeduplicationId'] = md5(json_encode($params));
+        }
 
-        Log::info("{$event} sent to bus queue", [
-            'message_id' => $response->get('MessageId'),
-            'message' => $message,
-        ]);
+        try {
+            $response = $this->sqs->sendMessage($payloadSqs);
+
+            $eventName = $params['events'][0];
+
+            Log::info("{$eventName} sent to bus queue", [
```

Comment:
is there an opportunity for this bit to respect the dontreport config option too? :D

Reply:
scratch this wont make sense.. we need another config to determine the loglevel or quiet this donw altogether.. dontreport is used for something else entirely